tests/generative/test_query_model.py

```diff
@@ -89,7 +89,7 @@
     # In development we never want to run with derandomize because it's
     # less effective at finding bugs and, more importantly, has very slow
     # replay due to turning off the test database
-    derandomize=bool(os.environ.get("GENTEST_RANDOMIZE"))
+    derandomize=(not os.environ.get("GENTEST_RANDOMIZE"))
```

Comment:
This will do what we want in CI, but only for the long-running generative tests, because `GENTEST_RANDOMIZE` is set in the `generative-tests.yml`. It isn't set for the subset of tests run with the standard `just test-all` in the main.yaml workflow, so presumably it is using derandomise there. It seems a bit weird to set it to the opposite of whatever it's set to when we're running in one specific CI workflow, and it means it's not doing what the comment says, and never running with derandomize in development (that's maybe just a typo in the comment which should say CI).  
Also the double-negative is making my head hurt, so I'm not convinced I've got the true/false-ness the right way round there anyway.

Reply:
I agree the double-negative is weird here, but I think the behaviour is what we want, which is: never derandomize in development, always derandomize in CI unless explicitly told not to be the long-running tests.
```
python -c '\
for CI in ("t", None):
    for GENTEST_RANDOMIZE in ("t", None):
        derandomize=(not GENTEST_RANDOMIZE) if CI else False
        print(f"{CI=}, {GENTEST_RANDOMIZE=}, {derandomize=}")'
```

```
CI='t', GENTEST_RANDOMIZE='t', derandomize=False
CI='t', GENTEST_RANDOMIZE=None, derandomize=True
CI=None, GENTEST_RANDOMIZE='t', derandomize=False
CI=None, GENTEST_RANDOMIZE=None, derandomize=False
```

Reply:
So we _do_ want to derandomize for the non-long-running generative tests?  Maybe it's the comment that's confusing then.  Should we add something to the developer docs to point out that in non-CI, this env variable has no effect? Currently there are docs that tell you that you can use it locally and suggest that you should when you add new query model operations or change query engines

Reply:
Uh yeah you're right, the docs are misleading then, and also the behaviour is fundamentally confusing. I think maybe a better thing would be to invert this completely and have a `GENTEST_DERANDOMIZE` variable that we set explicitly in `just test-all` and then we don't need to set it anywhere else. That would be much simpler.